org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/PersistentModel.java

```diff
@@ -9,76 +9,41 @@
 package org.eclipse.buildship.core.preferences;
 
 import java.util.Collection;
+import java.util.List;
 
 import com.google.common.base.Optional;
 
+import org.eclipse.core.resources.IFolder;
+import org.eclipse.core.resources.IResource;
 import org.eclipse.core.runtime.IPath;
+import org.eclipse.jdt.core.IClasspathEntry;
 
 /**
- * Interface to load and store Gradle model elements stored in the workspace plugin state area. The
- * model elements can be stored and retrieved in either a structured or in a map-like fashion.
+ * Interface to load and store Gradle model elements stored in the workspace plugin state area.
+ * <p/>
+ * The {@link Optional} return type on the getters mark whether a value is present in the model.
  *
  * @author Donat Csikos
  */
 public interface PersistentModel {
 
-    /**
-     * Returns the build directory.
-     * @return the build directory
-     */
     Optional<IPath> getBuildDir();
```

Comment:
Personally I don't find Optional very useful as long as we're not on Java 8 and heavily using functional idioms. Also, if we ever open this up to others (which there are good use cases for), we'd have to remove the Guava types from the signatures anyway so we don't leak the dependency. Let's stick with `null` and reconsider this once our minimum version is Java 8.